communication-service/index.js

```diff
@@ -28,6 +30,27 @@ io.on('connection', (socket) => {
       socket.to(roomId).emit('receive-chat', senderUsername, chatMessage);
     }
   });
+
+  // leave-session: when user clicks on "leave session" button
+  // disconnecting: when user closes tab/disconnects
+  const sessionEndEvents = ['leave-session', 'disconnecting'];
+
+  for (const event of sessionEndEvents) {
+    socket.on('leave-session', () => {
+      socket.rooms.forEach((room) => {
+        if (room !== socket.id) {
+          // Emit to other sockets in the same room this socket had joined
```

Comment:
This piece of code seems similar to the one in collaboration-service. If we have time, we can abstract it. But for now, this works!

Reply:
I think it's ok to be the same since they are doing the same thing?

Reply:
But yea will abstract these socket event listener out in future PR, thanks for pointing out.

Reply:
Alright sure!